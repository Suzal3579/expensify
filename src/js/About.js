import React from 'react';
import { Header, List } from 'semantic-ui-react';

const About = () => (
    <div>
        <Header color="grey">About</Header>
        <p>
            So, some of your friends/ relatives owe you some money? <br />
            If yes, then you are not alone. This happens to every one of us.
            We lend people some money and later forget to take it back. This app
            takes information about your money and even keeps a date including
            some handy notes to remember later. Now all you have to do is remember
            to open this app and all your money details/ expenditure are available
            right on your screen.
        </p>
        <Header color="grey">Developer</Header>
        <List>
            <List.Item icon="users" content="Pramesh Bajracharya" />
            <List.Item icon="marker" content="Kathmandu, Nepal" />
            <List.Item icon="mail" content={<a href="mailto:pe.messh@gmail.com">pe.messh@gmail.com</a>} />
            <List.Item icon="linkify" content={<a href="http://bajracharya.me">My website and blogs.</a>} />
        </List>
    </div>
);

export default About;
