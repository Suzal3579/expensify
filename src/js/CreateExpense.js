import React from 'react';
import { connect } from 'react-redux';

import { addExpense } from '../actions/expenses';

// eslint-disable-next-line
const CreateExpense = ({ dispatch }) => (
    <div>
        <h1>Create Expense</h1>
        <form onSubmit={(e) => {
            e.preventDefault();
            dispatch(addExpense({
                description: e.target.description.value,
                note: e.target.note.value,
                amount: e.target.amount.value,
            }));
            e.target.description.value = '';
            e.target.amount.value = '';
            e.target.note.value = '';
        }}
        >
            <input type="text" name="description" />
            <input type="text" name="note" />
            <input type="text" name="amount" />
            <button type="submit" >Add</button>
        </form>
    </div>
);

export default connect()(CreateExpense);
