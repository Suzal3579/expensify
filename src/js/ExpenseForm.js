import React, { Component } from 'react';
import 'react-dates/initialize';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? props.expense.amount : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            errorState: '',
        };
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    }
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    }
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    }
    onDateChange = (createdAt) => {
        if (createdAt) { this.setState(() => ({ createdAt })); }
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    }
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({
                errorState: 'Sorry but you cannot leave these fields empty ...',
            }));
        } else {
            this.setState(() => ({
                errorState: '',
            }));
            this.props.onSubmit({
                description: this.state.description,
                note: this.state.note,
                amount: this.state.amount,
                createdAt: this.state.createdAt.valueOf(),
            });
        }
    }
    render() {
        return (
            <div>
                {this.state.errorState && <p>{this.state.errorState}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        value={this.state.description}
                        placeholder="Description"
                        onChange={this.onDescriptionChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                    <br />
                    <input
                        type="text"
                        value={this.state.note}
                        placeholder="Note"
                        onChange={this.onNoteChange}
                    />
                    <input
                        type="text"
                        value={this.state.amount}
                        placeholder="Amount"
                        onChange={this.onAmountChange}
                    />
                    <button type="submit">Add</button>
                </form>
            </div>
        );
    }
}

