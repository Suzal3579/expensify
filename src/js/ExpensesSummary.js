import React from 'react';
import { connect } from 'react-redux';

import getVisibleExpenses from '../apis/getVisibleExpenses';
import expensesTotal from '../apis/expensesTotal';

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
    const wordToDisplay = expenseCount === 1 ? 'expense' : 'expenses';
    return (
        <div>
            {expenseCount > 0 ?
                <h1>{expenseCount} {wordToDisplay} totalling : {expenseTotal}</h1> : undefined
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: expensesTotal(visibleExpenses),
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
