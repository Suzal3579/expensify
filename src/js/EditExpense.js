import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button } from 'semantic-ui-react';

import firebase, { database } from '../firebase/firebase';
import { removeExpense, editExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpense extends Component {
    onSubmit = (expense) => {
        // Updating the data ...
        const currentUserUid = firebase.auth().currentUser.uid;
        database.ref(`users/${currentUserUid}/expenses`)
            .child(expense.id)
            .update({
                description: expense.description,
                note: expense.note,
                amount: expense.amount,
                createdAt: expense.createdAt,
            })
            .then(() => {
                this.props.history.push('/');
            });
    }
    onClick = () => {
        // Removing the item from firebase ...
        const currentUserUid = firebase.auth().currentUser.uid;
        database.ref(`users/${currentUserUid}/expenses`)
            .child(this.props.match.params.id)
            .remove(() => {
                this.props.history.push('/');
            });
    }
    render() {
        return (
            <div>
                <div className="col-md-6 col-md-offset-3">
                    <Button
                        fluid
                        negative
                        onClick={this.onClick}
                    >
                        Remove
                    </Button>
                    <br /><br />
                </div>
                <ExpenseForm
                    id={this.props.match.params.id}
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find(expense => expense.id === props.match.params.id),
});
const mapDispatchToProps = dispatch => ({
    editExpense: (id, expense) => { dispatch(editExpense(id, expense)); },
    removeExpense: (id) => { dispatch(removeExpense(id)); },
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
