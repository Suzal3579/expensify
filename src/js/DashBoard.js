import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const DashBoard = () => (
    <div>
        <h1>DashBoard</h1>
        <ExpenseListFilters />
        <ExpenseList />
        <ExpensesSummary />
    </div>
);

export default DashBoard;
