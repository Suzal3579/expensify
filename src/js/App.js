import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';

import Header from './Header';
import DashBoard from './DashBoard';
import CreateExpense from './CreateExpense';
import Help from './Help';
import NoMatch from './NoMatch';
import store from '../store/store';
import { addExpense } from '../actions/expenses';
import { setTextFilter } from '../actions/filters';

const storeInstance = store();

storeInstance.dispatch(addExpense({
    description: 'Suzal is getting good at this.',
    amount: 90000,
    note: 'Team Lead',
    createdAt: 234521,
}));
storeInstance.dispatch(addExpense({
    description: 'Pramesh is getting good at many things.',
    amount: 190000,
    note: 'CTO',
    createdAt: 102123,
}));
storeInstance.dispatch(setTextFilter('Pramesh'));
setTimeout(() => {
    storeInstance.dispatch(setTextFilter('Suzal'));
}, 2000);

const App = () => (
    <Provider store={storeInstance}>
        <BrowserRouter>
            <div>
                <Header />
                <Switch>
                    <Route exact path="/" component={DashBoard} />
                    <Route path="/dashboard/:id" component={DashBoard} />
                    <Route path="/createExpense" component={CreateExpense} />
                    <Route path="/help" component={Help} />
                    <Route component={NoMatch} />
                </Switch>
            </div>
        </BrowserRouter>
    </Provider>
);

export default App;
