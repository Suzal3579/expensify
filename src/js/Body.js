import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Sidebar, Segment, Menu } from 'semantic-ui-react';

import DashBoard from './DashBoard';
import CreateExpense from './CreateExpense';
import EditExpense from './EditExpense';
import Help from './Help';
import MenuItems from './MenuItems';
import NoMatch from './NoMatch';

const Body = props => (
    <Sidebar.Pushable as={Segment}>
        <Sidebar
            as={Menu}
            animation="scale down"
            width="thin"
            direction="right"
            visible={props.visible}
            icon="labeled"
            vertical
            inverted
        >
            <MenuItems />
        </Sidebar>
        <Sidebar.Pusher>
            <Segment basic>
                <div id="body">
                    <Switch>
                        <Route exact path="/" component={DashBoard} />
                        <Route path="/createExpense" component={CreateExpense} />
                        <Route path="/editExpense/:id" component={EditExpense} />
                        <Route path="/help" component={Help} />
                        <Route component={NoMatch} />
                    </Switch>
                </div>
            </Segment>
        </Sidebar.Pusher>
    </Sidebar.Pushable>
);

export default Body;
