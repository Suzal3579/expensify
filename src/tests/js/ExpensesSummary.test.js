import { ExpensesSummary } from '../../js/ExpensesSummary';
import { shallow } from 'enzyme';
import React from 'react';

it('should properly render when one expense is there', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={200} />);
    expect(wrapper).toMatchSnapshot();
});

it('should properly render when one expense is there', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={100} expensesTotal={80190918200} />);
    expect(wrapper).toMatchSnapshot();
})