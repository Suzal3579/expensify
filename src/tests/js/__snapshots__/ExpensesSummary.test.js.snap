// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render when one expense is there 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    expenseCount={1}
    expensesTotal={200}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        1
         
        expense
         totalling : 
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          1,
          " ",
          "expense",
          " totalling : ",
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        1,
        " ",
        "expense",
        " totalling : ",
        undefined,
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          1
           
          expense
           totalling : 
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            " ",
            "expense",
            " totalling : ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          1,
          " ",
          "expense",
          " totalling : ",
          undefined,
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should properly render when one expense is there 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    expenseCount={100}
    expensesTotal={80190918200}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        100
         
        expenses
         totalling : 
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          100,
          " ",
          "expenses",
          " totalling : ",
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        100,
        " ",
        "expenses",
        " totalling : ",
        undefined,
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          100
           
          expenses
           totalling : 
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            100,
            " ",
            "expenses",
            " totalling : ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          100,
          " ",
          "expenses",
          " totalling : ",
          undefined,
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
